core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -52,7 +55,43 @@ private[spark] class TaskDescription(
     val addedFiles: Map[String, Long],
     val addedJars: Map[String, Long],
     val properties: Properties,
-    val serializedTask: ByteBuffer) {
+    private var serializedTask_ : ByteBuffer) extends  Logging {
```

Comment:
ah, that last one is doing the full refactor and fixing the 100+ compile errors :)

Here is yet another version, I think more like what Kay was suggesting: https://github.com/squito/spark/commit/389fec55de99e9f12104a8ba2b7aa54278aadd93

I agree with Kay that the `serializedTask` member var is a little confusing in the version in this pr now.  No strong preference between the alternatives; the full separation is cleanest, but the change is pretty big (from a quick glance, I'm not sure all those test changes are correct, but I didn't look closely).